Bottom: 1bd5004fa8dc28957a12ee898146c01475658b33
Top:    e12e2a65768bec2151d832fa45e6139d7e0f0354
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-09-30 17:08:18 +0800

Refresh of introduce-core-ring

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 3d57e22..89384ce8 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1868,7 +1868,7 @@ static ssize_t tun_put_user_xdp(struct tun_struct *tun,
 	u64_stats_update_end(&stats->syncp);
 	put_cpu_ptr(tun->pcpu_stats);
 
-	return 0;
+	return size + vnet_hdr_sz;
 }
 
 /* Put packet to the user space buffer */
diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 51ce939..66c8217 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -831,6 +831,7 @@ static void handle_rx(struct vhost_net *net)
 			pr_debug("Discarded rx packet: "
 				 " len %d, expected %zd\n", err, sock_len);
 			vhost_discard_vq_desc(vq, headcount);
+			printk("err %d sock_len %d\n", err, sock_len);
 			continue;
 		}
 		/* Supply virtio_net_hdr if VHOST_NET_F_VIRTIO_NET_HDR */
